frontend/src/app/auth/auth.service.ts

```diff
@@ -8,20 +14,42 @@ export class AuthService {
 
   public redirectedFrom: string;
 
-  constructor(private router: Router) {
-  }
+  private userData$: Subject<User> = new ReplaySubject<User>();
 
-  isLoggedIn(): boolean {
-    return !!localStorage.getItem('user');
+  constructor(private router: Router, private httpClient: HttpClient) {
+    this.fetchUserDataIfNeeded();
   }
 
-  login(mail: string, password: string): void {
-    localStorage.setItem('user', mail);
-    this.router.navigate(this.redirectedFrom ? [this.redirectedFrom] : ['/']);
+  public login(login: string, password: string): void {
+    this.httpClient.post<User>('http://localhost:8080/users/login', { login, password })
```

Comment:
You  could save this as a global constant: "http://localhost:8080"? Later it will be easier to change ports/protocol etc.